### DISCUSSION TITLE
Typing for `add_middleware` breaks with suggested CORS config.

### DISCUSSION THREAD

**Discussion by ArzelaAscoIi** [encode/starlette#2423] [Potential Issue] [answered]
Hi! We are using the `add_middleware` for CORS like this: 
```py
application.add_middleware(
      CORSMiddleware,
      allow_origins=["*"],  
      allow_credentials=False, 
      allow_methods=["*"],  
      allow_headers=["*"], 
  )
```

After updating starlette, we got this error from mypy

`Argument "allow_origins" to "add_middleware" of "Starlette" has incompatible type "list[str]"; expected <nothing>`

Its somehow related to this PR: https://github.com/encode/starlette/pull/2381

Any Idea on how to fix this ?

**Comment by Kludex** [comment]
What's the mypy version? I can't reproduce that.

**Reply by ArzelaAscoIi** [reply]
We are using mypy==1.5.1. Let me upgrade and see if it fixes it 🙈

**Reply by ArzelaAscoIi** [reply, marked as answer]
argh. you are right. 1.8.0 fixed it. Sorry and thanks for your help!
